### DISCUSSION TITLE
消息轨迹希望支持更细粒度的场景

### DISCUSSION THREAD

**Discussion by shirenchuang** [apache/rocketmq#7139] [General]
目前开启消息轨迹，一般都是生产者实例和消费者实例在构造函数的时候指定了，所以只要是通过他们发送和消费的消息都会发送轨迹消息；

但是这样会有一个问题，QPS大的情况下，消息轨迹的写入压力也会很大，因为相当于是将业务的消息 Double了一下；

但是大部分场景下，很多消息都不太关注消息轨迹，只有部分消息需要关注；

我觉得可以支持一下更细粒度的配置

在Producer和Consumer都开启了消息轨迹的前提下

新增一个配置，用来 Match 哪些Topic、Tag 的需要轨迹消息 ; 

只有满足匹配条件下，才需要发送轨迹消息；

这个改动量不大。

At present, when the message track is enabled, the producer instance and the consumer instance are generally specified in the constructor, so as long as the message is sent and consumed through them, the track message will be sent;

But there will be a problem in this way. When the QPS is large, the writing pressure of the message track will also be very high, because it is equivalent to double the business message;

However, in most scenarios, many messages do not pay much attention to the message track, and only some messages need attention;

I think it can support more fine-grained configuration

On the premise that both Producer and Consumer have opened the message track

Add a configuration to match which Topics and Tags need track messages;

Only when the matching conditions are met, the trajectory message needs to be sent;

This change is not large.

**Comment by joeCarf** [comment]
I'd like to try, you can start an issue and assign it to me.
